invenio_app_ils/ui/src/invenio_app_ils/pages/backoffice/ItemDetails/state/actions.js

```diff
@@ -25,3 +37,36 @@ export const fetchItemDetails = itemPid => {
       });
   };
 };
+
+export const deleteItem = (itemPid, ownProps) => {
+  return async dispatch => {
+    dispatch({
+      type: DELETE_IS_LOADING,
+    });
+
+    await itemApi
+      .del(itemPid)
+      .then(response => {
```

Comment:
Did we decide yet if we're gonna use `async` or promises API?

If we use `async` it's probably cleaner if we do: `const response = await itemApi.del(itemPid);`

Reply:
We should also wrap it in a try catch

Reply:
Also here :)